pandas/core/groupby.py

```diff
@@ -3531,6 +3531,13 @@ def transform(self, func, *args, **kwargs):
         Each subframe is endowed the attribute 'name' in case you need to know
         which group you are working on.
 
+        The current implementation imposes two requirements on f:
+
+        * f must return a new DataFrame - in-place mutation of subframes is
+          not supported and may produce unexpected results.
+        * f must support application both column-by-column and to the entire
+          subframe.
```

Comment:
Also this is not entirely true I think. In any case it does work (testing with a simple example) with functions that only work on a Series (so column by column)